### PR TITLE
Minimal cookiecutter output

### PR DESCRIPTION
**Pull request by kotfic**
Transitioning this from fork. See #107

### CODE REVIEW

**Comment by kotfic** [comment]
Adding a test for the minimal install revealed some linting errors that have been addressed.  Two potentially controversial things to consider:

1. This introduces the `pytest-custom_exit_code` plugin that suppresses errors if there are  no tests. This is the case with the minimal install.
2. Adds `# noqa: F401` statements to suppress warnings from unused imports in the case of a minimal install. Including these imports (to me) seems useful in terms of directing what should be happening in these files. Alternately I can exclude them and leave empty files if this approach is unappealing.

File: {{ cookiecutter.project_slug }}/tox.ini [hunk 1]
```diff
@@ -50,8 +50,10 @@ deps =
     pytest-django
     pytest-factoryboy
     pytest-mock
+    # Note: implements --suppress-no-test-exit-code. Pass if there are no tests.
+    pytest-custom_exit_code
 commands =
-    pytest {posargs}
+    pytest --suppress-no-test-exit-code {posargs}
```
Comment (brianhelba) [inline {{ cookiecutter.project_slug }}/tox.ini:56]: CLI arguments to pytest should generally go in the `[pytest] addopts` section below.

However, this is obsoleted by #113, if we accept that.

**Comment by brianhelba** [comment]
@kotfic I pushed some hopefully-uncontroversial changes directly on top of this branch. Let me know if you approve.

**Comment by brianhelba** [comment]
@kotfic As for your own more "controversial" things, I made #113 and #114 to hopefully resolve them in a better way. Please let me know what you think.

If you approve, those PRs will merge into this branch.

File: .circleci/config.yml [hunk 2]
```diff
@@ -30,11 +30,43 @@ jobs:
             DJANGO_MINIO_STORAGE_ACCESS_KEY: minioAccessKey
             DJANGO_MINIO_STORAGE_SECRET_KEY: minioSecretKey
           working_directory: my-new-project
+
+  test-minimal:
```
Comment (brianhelba) [inline .circleci/config.yml:34]: When we (soon) switch to GitHub Actions, this should be much less verbose to do as a matrix.

**Review by brianhelba — APPROVED** [review]
@kotfic Great feature, thanks for all your work, particularly to get the new tests passing.

As soon as you approve my additions and we resolve #113 and #114 in some way, this should be good to merge.

**Comment by kotfic** [comment]
No objections on my part.  Looks good.  Pull the lever

**Pull request merged**